Source/Core/Core/AchievementManager.cpp

```diff
@@ -212,6 +214,43 @@ AchievementManager::ResponseType AchievementManager::FetchGameData()
       rc_api_process_fetch_game_data_response);
 }
 
+void AchievementManager::ActivateDeactivateAchievement(AchievementId id, bool enabled,
+                                                       bool unofficial, bool encore)
+{
+  UnlockStatus status = m_unlock_map[id];
```

Comment:
`const UnlockStatus& status`. Also, is it intentional that you're creating a new entry here if one doesn't exist yet? You may want `m_unlock_map.find(id)` instead.

Reply:
You mean m_unlock_map.at(id)? .find() returns an iterator.